client/src/components/shared/Select.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * A generic Select component meant as a replacement for
+ * the select tag.
+ */
+const StyledSelect = styled.select`
+  margin: 0;
+  border-radius: 0;
+  text-align-last: left;
+  padding-left: var(--space-2);
+  padding-top: var(--space-2);
+  padding-bottom: var(--space-2);
+  width: ${(props) => props.width};
+`;
+
+function Select({
+  value,
+  onChange,
+  label,
+  options,
+  selectionPrompt,
+  disabled,
+  ...props
+}) {
+  return (
```

Comment:
Seems like this is only super valuable if it gets used a lot but I like the idea of it

Reply:
Select inputs form a significant part of the specs for one of my subcomponents, so I made `Select` to at least incorporate all of the functionality that I need for them. Ideally whenever we're using a select input on the page we'd use `Select` and just add a variant if the styling doesn't suit the situation we need it for. In general the goal of these generic components is to have a single source of truth for the styling and functionality of a particular element.